src/api/app/components/workflow_run_filter_link_component.rb

```diff
@@ -0,0 +1,38 @@
+class WorkflowRunFilterLinkComponent < ApplicationComponent
+  def initialize(text:, filter_item:, selected_filter:, token:, amount: 0)
+    super
+
+    @text = text
+    @filter_item = filter_item
+    @selected_filter = selected_filter
+    @amount = ensure_integer_amount(amount)
+    @token = token
+  end
+
+  def css_for_link
+    workflow_run_filter_matches? ? 'active' : ''
+  end
+
+  def css_for_badge_color
+    workflow_run_filter_matches? ? 'badge-light' : 'badge-primary'
+  end
+
+  private
+
+  def workflow_run_filter_matches?
+    if @selected_filter[:status].present?
+      @filter_item[:status] == @selected_filter[:status]
+    elsif @selected_filter[:event_type].present?
+      @filter_item[:event_type] == @selected_filter[:event_type]
+    elsif @selected_filter.empty?
+      @filter_item.empty?
+    end
+  end
+
+  # This method won't be needed in Ruby 2.6+ by using the option `exception: true` for Integer(...), it's then a one-liner
+  def ensure_integer_amount(amount)
```

Comment:
@danidoni is this really necessary? on all items you did a `count` in the finder class, dont see how this should be a string :)

Reply:
Hmm, let me check

Reply:
True, removed